src/control/client/config_test.go

```diff
@@ -0,0 +1,71 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package client
+
+import (
+	. "github.com/daos-stack/daos/src/control/common"
+	"os"
+	"testing"
+)
+
+const (
+	testRuntimeDir = "/var/tmp/runtime"
+)
+
+func TestParseEnvVars(t *testing.T) {
+	// Obtain the default client configuration
+	config := NewConfiguration()
+
+	// Save the environment variable before we blow it away
+	original := config.ext.getenv(daosAgentDrpcSockEnv)
```

Comment:
This works fine for this test case, but the great advantage of creating an interface for external dependencies is that you can mock them. I.e. You don't have to mangle your real environment or talk to the real OS. If you haven't used that technique much, do feel free to talk to me or Tom about it.